src/lib.rs

```diff
@@ -162,9 +162,7 @@ pub fn bn128_pairing(input: &[u8], output: &mut [u8; 32]) -> Result<(), Error> {
             vals.push((a, b));
         }
 
-        let mul = vals
-            .into_iter()
-            .fold(Gt::one(), |s, (a, b)| s * pairing(a, b));
+        let mul = pairing_batch(&vals);
```

Comment:
Do you want to upstream this to parity also?

Reply:
they already did https://github.com/paritytech/parity-ethereum/pull/10765

Reply:
Ah ok! If you said it was from upstream i wouldn't have had any questions :)